src/CsvHelper/Configuration/ConfigurationFunctions.cs

```diff
@@ -16,35 +16,37 @@ namespace CsvHelper.Configuration
 	public static class ConfigurationFunctions
 	{
 		private static readonly char[] lineEndingChars = new char[] { '\r', '\n' };
+		private static readonly string[] lineEndings = new[] { "\r\n", "\r", "\n" };
+		private static readonly Regex specialCharsRegex = new Regex(@"([.$^{\[(|)*+?\\])");
 
 		/// <summary>
 		/// Throws a <see cref="ValidationException"/> if <see name="HeaderValidatedArgs.InvalidHeaders"/> is not empty.
 		/// </summary>
 		public static void HeaderValidated(HeaderValidatedArgs args)
 		{
-			if (args.InvalidHeaders.Count() == 0)
+			if (!args.InvalidHeaders.Any())
 			{
 				return;
 			}
 
 			var errorMessage = new StringBuilder();
 			foreach (var invalidHeader in args.InvalidHeaders)
 			{
-				errorMessage.AppendLine($"Header with name '{string.Join("' or '", invalidHeader.Names)}'[{invalidHeader.Index}] was not found.");
+				errorMessage.Append("Header with name '").Append(string.Join("' or '", invalidHeader.Names)).Append("'[").Append(invalidHeader.Index).AppendLine("] was not found.");
```

Comment:
Note, perhaps it saves a string allocation on tfm < net6.0, but it is not an optimisation on net6.0 (see https://devblogs.microsoft.com/dotnet/string-interpolation-in-c-10-and-net-6/)

Reply:
I could use conditional compilation but I don't think having this for .NET 6, makes performance worse.

The example in your link:

```cs
public static void AppendVersion(StringBuilder builder, int major, int minor, int build, int revision) =>
    builder.Append($"{major}.{minor}.{build}.{revision}");
```

essentially gets transformed by compiler to:

```cs
public static void AppendVersion(StringBuilder builder, int major, int minor, int build, int revision)
{
    var handler = new AppendInterpolatedStringHandler(3, 4, builder);
    handler.AppendFormatted(major);
    handler.AppendLiteral(".");
    handler.AppendFormatted(minor);
    handler.AppendLiteral(".");
    handler.AppendFormatted(build);
    handler.AppendLiteral(".");
    handler.AppendFormatted(revision);
    builder.Append(ref handler);
}
```

Reply:
Related: I was thinking of using `AppendJoin` initially but I believe it is not supported in one of the .NET Framework versions listed in `TargetFrameworks`.

I had to change `<TargetFrameworks>` in the `csproj` file locally because I only have .NET 6+7 + .Net Framework 4.8 on my machine 😃

Reply:
Agree it makes the same thing on net6.0, I did not mean it was a de-optimisation but rather it is arguably sacrificing readability for negligible performance benefit - this method would normally only get called once (after `ReadHeader`) and at this point we are about to throw an exception. Either way it looks fine to me

> but I believe it is not supported in one of the .NET Framework versions listed in TargetFrameworks.

Yes sadly a common occurrence these days